### PR TITLE
Improve plugin config & mediapipe init logging

### PR DESCRIPTION
**Pull request by atobiszei**
Improve plugin config & mediapipe init logging

* Log sorted by name for easier debug & audit
* Don't flood OVMS log with spdlog format (20kB vs 53kB per OVMS start, 236MB
vs 248MB per unit tests run)
* Unify logging plugin config code between ModelManager and ModelInstance
* Fix ov tracing logs appearing on non-debug builds

### CODE REVIEW

File: src/ov_utils.hpp [hunk 1]
```diff
@@ -40,4 +44,45 @@ std::optional<ov::Layout> getLayoutFromRTMap(const ov::RTMap& rtMap);
...
+template <typename PropertyExtractor>
+static void logOVPluginConfig(PropertyExtractor&& propertyExtractor, const std::string& loggingAuthor, const std::string& loggingDetails) {
+    SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Logging {}; {}plugin configuration", loggingAuthor, loggingDetails);
+    auto key = std::string("SUPPORTED_PROPERTIES");  // ov::supported_properties;
+    std::vector<ov::PropertyName> supportedConfigKeys;
+    try {
+        ov::Any value = propertyExtractor(key);
+        supportedConfigKeys = value.as<std::vector<ov::PropertyName>>();
+    } catch (std::exception& e) {
+        SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Exception thrown from OpenVINO when requesting {};{} config key: {}; Error: {}", loggingAuthor, loggingDetails, key, e.what());
```
Comment (michalkulakowski) [inline src/ov_utils.hpp:61]:
```suggestion
        SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Exception thrown from OpenVINO when requesting {};{}config key: {}; Error: {}", loggingAuthor, loggingDetails, key, e.what());
```

File: src/ov_utils.hpp [hunk 2]
```diff
@@ -40,4 +44,45 @@ std::optional<ov::Layout> getLayoutFromRTMap(const ov::RTMap& rtMap);
...
+        return;
+    } catch (...) {
+        SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Exception thrown from OpenVINO when requesting {};{} config key: {};", loggingAuthor, loggingDetails, key);
```
Comment (michalkulakowski) [inline src/ov_utils.hpp:64]:
```suggestion
        SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Exception thrown from OpenVINO when requesting {};{}config key: {};", loggingAuthor, loggingDetails, key);
```

File: src/ov_utils.hpp [hunk 3]
```diff
@@ -40,4 +44,45 @@ std::optional<ov::Layout> getLayoutFromRTMap(const ov::RTMap& rtMap);
...
+    std::vector<std::string> pluginConfigNameValues;
+    for (auto& key : supportedConfigKeys) {
+        if (key == "SUPPORTED_PROPERTIES")
+            continue;
+        std::string value;
+        try {
+            ov::Any paramValue = propertyExtractor(key);
+            value = paramValue.as<std::string>();
+        } catch (std::exception& e) {
+            SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Exception thrown from OpenVINO when requesting {};{} config key: {}; Error: {}", loggingAuthor, loggingDetails, key, e.what());
```
Comment (michalkulakowski) [inline src/ov_utils.hpp:76]:
```suggestion
            SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Exception thrown from OpenVINO when requesting {};{}config key: {}; Error: {}", loggingAuthor, loggingDetails, key, e.what());
```

File: src/ov_utils.hpp [hunk 4]
```diff
@@ -40,4 +44,45 @@ std::optional<ov::Layout> getLayoutFromRTMap(const ov::RTMap& rtMap);
...
+            continue;
+        } catch (...) {
+            SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Exception thrown from OpenVINO when requesting {};{} config key: {};", loggingAuthor, loggingDetails, key);
```
Comment (michalkulakowski) [inline src/ov_utils.hpp:79]:
```suggestion
            SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Exception thrown from OpenVINO when requesting {};{}config key: {};", loggingAuthor, loggingDetails, key);
```

**Review by michalkulakowski — APPROVED** [review]

**Review by mzegla — APPROVED** [review]

**Review by rasapala — APPROVED** [review]
LGTM

File: src/ov_utils.hpp (hunk 1 above)
Comment (atobiszei) [reply src/ov_utils.hpp:63]: I changed format in modelinstance.cpp now when triggered by requesting non existing key:
[2024-01-03 10:24:39.717][402856][modelmanager][debug][ov_utils.hpp:77] Exception thrown from OpenVINO when requesting OpenVINO Core plugin: GNA; config key: INFERENCE_PRECISION_HINT; Error: Exception from src/inference/src/core.cpp:228:
[ GENERAL_ERROR ] [openvino_intel_gna_plugin] in function GetMetric: Unsupported parameters for GetMetric: ABECADLO

[2024-01-03 10:24:39.734][402856][modelmanager][debug][ov_utils.hpp:77] Exception thrown from OpenVINO when requesting compiled model: dummy; version: 1; target device: CPU; config key: OPTIMAL_NUMBER_OF_INFER_REQUESTS; Error: Exception from src/inference/src/compiled_m
odel.cpp:119:
Exception from src/plugins/intel_cpu/src/compiled_model.cpp:193:
Unsupported property: ABECADLO

**Pull request merged**